internal/statetransition/state_transition.go

```diff
@@ -1168,8 +1168,8 @@ func calculateNewCoreAssignments(
 	validatorState validator.ValidatorState,
 	newTimeslot jamtime.Timeslot,
 	entropyPool state.EntropyPool,
-) state.CoreAssignments {
-	newAssignments := intermediateAssignments
+) (newAssignments state.CoreAssignments, reporters []ed25519.PublicKey) {
```

Comment:
Should we use a map instead to assure uniqueness?

Reply:
Which is exactly what the GP says, I think. `We note that the Ed25519 key of each validator whose
signature is in a credential is placed in the reporters set R`

Reply:
it's also hard to do because we use a slice instead of array and it is not comparable, but I can refactor that